coremltools/converters/mil/frontend/torch/test/test_torch_ops.py

```diff
@@ -2523,6 +2523,54 @@ def forward(self, x):
 
         self.run_compare_torch(input_shape, model, backend=backend)
 
+    @pytest.mark.parametrize("backend", backends)
+    def test_expand_dynamic_shape0(self, backend):
+        class TestModel(nn.Module):
+            def forward(self, x):
+                return x.expand(x.shape[1], x.shape[1])
+
+        self.run_compare_torch(torch.arange(20).reshape((1, 20)),
+                               TestModel(),
+                               input_as_shape=False,
+                               converter_input_type=[TensorType(shape=[1, ct.RangeDim()])],
+                               backend=backend)
+
+    @pytest.mark.parametrize("backend", backends)
+    def test_expand_dynamic_shape1(self, backend):
+        class TestModel(nn.Module):
+            def forward(self, x):
+                return x.expand(x.shape[0], 1, x.shape[-1], x.shape[-1])
+
+        self.run_compare_torch(torch.arange(20).reshape((1, 20)),
+                               TestModel(),
+                               input_as_shape=False,
+                               converter_input_type=[TensorType(shape=[1, ct.RangeDim()])],
+                               backend=backend)
+
+    @pytest.mark.parametrize("backend", backends)
+    def test_expand_dynamic_shape2(self, backend):
+        class TestModel(nn.Module):
+            def forward(self, x):
+                return x.expand(x.shape[-1], 1, x.shape[-1], x.shape[-1])
+
+        self.run_compare_torch(torch.arange(20).reshape((1, 20)),
+                               TestModel(),
+                               input_as_shape=False,
+                               converter_input_type=[TensorType(shape=[1, ct.RangeDim()])],
+                               backend=backend)
+
+    @pytest.mark.parametrize("backend", backends)
+    def test_expand_dynamic_shape_and_broadcast(self, backend):
+        class TestModel(nn.Module):
+            def forward(self, x):
+                return x.expand(x.shape[0], 10)
+
+        self.run_compare_torch(torch.arange(20).reshape((20, 1)),
+                               TestModel(),
+                               input_as_shape=False,
+                               converter_input_type=[TensorType(shape=[ct.RangeDim(), 1])],
+                               backend=backend)
```

Comment:
Great tests.
Right now the target shape only comes from the input shape,
for instance `x.shape[0], x.shape[1]` ...
I think we can add a test that. the target shape is just from another input,
to make sure that we can handle the cases where the "target_shape and input_shape" are 100% independant
Like:
```
  @pytest.mark.parametrize("backend", backends)
     def test_expand_dynamic_shape_from_another_input(self, backend):
         class TestModel(nn.Module):
             def forward(self, x, y):
                 return x.expand(y[0], y[1])

         self.run_compare_torch([torch.arange(20).reshape((20, 1)), torch.Tensor([20, 20])]
                                TestModel(),
                                input_as_shape=False,
                                converter_input_type=[TensorType(shape=[ct.RangeDim(), 1])],
```

Reply:
ok. Done. The tests passes